code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/Optimize.java

```diff
@@ -62,60 +62,106 @@ public static void updatePropositions(
             @NonNull final List<DecisionScope> decisionScopes,
             @Nullable final Map<String, Object> xdm,
             @Nullable final Map<String, Object> data) {
-        if (OptimizeUtils.isNullOrEmpty(decisionScopes)) {
-            Log.warning(
-                    OptimizeConstants.LOG_TAG,
-                    SELF_TAG,
-                    "Cannot update propositions, provided list of decision scopes is null or"
-                            + " empty.");
+
+        Event event = updatePropositionsHelper(decisionScopes, xdm, data, null);
```

Comment:
Can we avoid having `updatePropositionsHelper`? What if we call the other method like so `updatePropositions(decisionScopes, xdm, data, null)`? Wouldn't that work?

Reply:
Example from Optimize iOS SDK 
https://github.com/adobe/aepsdk-optimize-ios/blob/dev-v5.0.2/Sources/AEPOptimize/Optimize%2BPublicAPI.swift#L27

Reply:
The previous update API uses the `MobileCore.dispatchEvent` method whereas the newer version will need to call `MobileCore.dispatchEventWithResponseCallback`.
I have extracted the common logic from both the APIs in `updatePropositionsHelper`

Reply:
Both `MobileCore.dispatchEvent` and `MobileCore.dispatchEventWithResponseCallback` do the same thing under the hood. Only difference is one accepts a callback, other doesn't. They can be used interchangeably

Reply:
https://github.com/adobe/aepsdk-core-android/blob/main/code/core/src/phone/java/com/adobe/marketing/mobile/MobileCore.java#L304

https://github.com/adobe/aepsdk-core-android/blob/main/code/core/src/phone/java/com/adobe/marketing/mobile/MobileCore.java#L343

Reply:
refactored the code